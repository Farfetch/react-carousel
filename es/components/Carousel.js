function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = _getPrototypeOf(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

import { CarouselProvider } from "../context";
import PropTypes from 'prop-types';
import React, { Component } from 'react';
import Styles from "./Carousel.css";
import cx from 'classnames';

var Carousel = /*#__PURE__*/function (_Component) {
  _inherits(Carousel, _Component);

  var _super = _createSuper(Carousel);

  function Carousel() {
    _classCallCheck(this, Carousel);

    return _super.apply(this, arguments);
  }

  _createClass(Carousel, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          className = _this$props.className,
          children = _this$props.children,
          isInfinite = _this$props.isInfinite,
          isRTL = _this$props.isRTL,
          itemsToShow = _this$props.itemsToShow,
          itemsToScroll = _this$props.itemsToScroll,
          onAfterChange = _this$props.onAfterChange,
          startItem = _this$props.startItem,
          otherProps = _objectWithoutProperties(_this$props, ["className", "children", "isInfinite", "isRTL", "itemsToShow", "itemsToScroll", "onAfterChange", "startItem"]);

      var providerProps = {
        isInfinite: isInfinite,
        isRTL: isRTL,
        itemsToShow: itemsToShow,
        onAfterChange: onAfterChange,
        itemsToScroll: itemsToScroll,
        startItem: startItem
      };
      return /*#__PURE__*/React.createElement(CarouselProvider, providerProps, /*#__PURE__*/React.createElement("div", _extends({
        className: cx(Styles.container, className)
      }, otherProps), children));
    }
  }]);

  return Carousel;
}(Component);

Carousel.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  isInfinite: PropTypes.bool,
  isRTL: PropTypes.bool,
  itemsToShow: PropTypes.number,
  onAfterChange: PropTypes.func,
  itemsToScroll: PropTypes.number,
  startItem: PropTypes.number
};
Carousel.defaultProps = {
  itemsToScroll: 1
};
export default Carousel;