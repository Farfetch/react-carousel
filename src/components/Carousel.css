.container {
    position: relative;
    width: 100%;
    overflow: hidden;
    touch-action: manipulation;
}
