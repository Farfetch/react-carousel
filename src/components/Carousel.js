import { BaseCarousel } from '#base';
import styled from '@emotion/styled';

const Carousel = styled(BaseCarousel)`
    position: relative;
    width: 100%;
    overflow: hidden;
    touch-action: manipulation;
`;

Carousel.displayName = 'Carousel';

export default Carousel;
