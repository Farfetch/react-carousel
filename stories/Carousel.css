.area {
    max-width: 100%;
}

.area-small {
    max-width: 50%;
}

.item {
    display: grid;
    align-items: center;
    justify-items: center;
    text-decoration: none;
    color: white;
}

.item p {
    font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
    font-size: 200px;
    font-weight: bold;
    line-height: 0;
}

.item > * {
    grid-area: 1 / 1 / 2 / 2;
}

.first {
    background-image: linear-gradient(120deg, #f6d365 0%, #fda085 100%);
}

.second {
    background-image: linear-gradient(120deg, #84fab0 0%, #8fd3f4 100%);
}

.third {
    background-image: linear-gradient(120deg, #a1c4fd 0%, #c2e9fb 100%);
}

.fourth {
    background-image: linear-gradient(to top, #fad0c4 0%, #ffd1ff 100%);
}

.fifth {
    background-image: linear-gradient(120deg, #d4fc79 0%, #96e6a1 100%);
}

.sixth {
    background-image: linear-gradient(120deg, #667eea 0%, #764ba2 100%);
}

.seventh {
    background-image: linear-gradient(120deg, #f093fb 0%, #f5576c 100%);
}
