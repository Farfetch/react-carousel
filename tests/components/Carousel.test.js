import { createSerializer } from 'enzyme-to-json';
import { shallow } from 'enzyme';
import React from 'react';

import Carousel from '../../src/components/Carousel';

beforeAll(() => {
    expect.addSnapshotSerializer(createSerializer());
});

describe('<Carousel /> props', () => {
    it('should render children and custom attrs correctly', () => {
        const tree = shallow(
            <Carousel className="foo" data-attr="bar">
                <div>I accept children.</div>
            </Carousel>
        );

        expect(tree).toMatchSnapshot();
    });
});
