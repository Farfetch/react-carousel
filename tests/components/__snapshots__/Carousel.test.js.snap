// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Carousel /> props should render children and custom attrs correctly 1`] = `
<CarouselProvider
  isInfinite={false}
  isRTL={false}
  itemsToScroll={1}
  itemsToShow={1}
  startItem={0}
>
  <div
    className="wrapper foo"
    data-attr="bar"
  >
    <div>
      I accept children.
    </div>
  </div>
</CarouselProvider>
`;
